myning/chapters/mine/mining_minigame.py

```diff
@@ -0,0 +1,84 @@
+import random
+from enum import Enum, auto
+from functools import lru_cache
+
+from textual.widgets import Static
+
+
+class MiningScore(Enum):
+    RED = auto()
+    ORANGE = auto()
+    YELLOW = auto()
+    GREEN = auto()
+
+
+COLORS = ["red1", "orange1", "yellow1", "green1", "yellow1", "orange1", "red1"]
+WIDTHS = [1, 1.5, 1.5, 2, 1.5, 1.5, 1]
+assert len(COLORS) == len(WIDTHS)
+assert sum(WIDTHS) == 10
+
+
+class MiningMinigame(Static):
+    def __init__(self, duration: int):
+        super().__init__()
+        self.duration = duration
+        self.segment_width = random.choice([2, 4, 6])
+        self.segment_widths = [int(width * self.segment_width) for width in WIDTHS]
+        self.width = sum(self.segment_widths)
+        self.cursor = random.randint(0, self.width - 1)
```

Comment:
Good. This was one thing I wanted to make sure we had to avoid macros

Reply:
There's a bug in this that I will push up a commit for. If we start at one of the edges and the direction is incorrectly randomly chosen, then the cursor will flicker between the left and right edges. I'm making it so that it starts just short of the edges to avoid the flickering behavior caused by using modulus